src/parser.yy

```diff
@@ -218,193 +218,293 @@ attach_point_def : attach_point_def ident    { $$ = $1 + $2; }
-     | "(" IDENT MUL MUL ")" expr %prec CAST    { $$ = new ast::Cast($2, true, true, $6, @1 + @5); }
-     | "(" expr "," vargs ")"                   {
-                                                  auto args = new ast::ExpressionList;
-                                                  args->emplace_back($2);
-                                                  args->insert(args->end(), $4->begin(), $4->end());
-                                                  $$ = new ast::Tuple(args, @$);
-                                                }
-     | pre_post_op                              { $$ = $1; }
-     ;
-
-pre_post_op : map_or_var INCREMENT   { $$ = new ast::Unop(token::INCREMENT, $1, true, @2); }
-            | map_or_var DECREMENT   { $$ = new ast::Unop(token::DECREMENT, $1, true, @2); }
-            | INCREMENT map_or_var   { $$ = new ast::Unop(token::INCREMENT, $2, @1); }
-            | DECREMENT map_or_var   { $$ = new ast::Unop(token::DECREMENT, $2, @1); }
-            | ident INCREMENT      { error(@1, "The ++ operator must be applied to a map or variable"); YYERROR; }
-            | INCREMENT ident      { error(@1, "The ++ operator must be applied to a map or variable"); YYERROR; }
-            | ident DECREMENT      { error(@1, "The -- operator must be applied to a map or variable"); YYERROR; }
-            | DECREMENT ident      { error(@1, "The -- operator must be applied to a map or variable"); YYERROR; }
-            ;
-
-ident : IDENT         { $$ = $1; }
-      | BUILTIN       { $$ = $1; }
-      | CALL          { $$ = $1; }
-      | CALL_BUILTIN  { $$ = $1; }
-      | STACK_MODE    { $$ = $1; }
-      ;
-
-call : CALL "(" ")"                 { $$ = new ast::Call($1, @$); }
-     | CALL "(" vargs ")"           { $$ = new ast::Call($1, $3, @$); }
-     | CALL_BUILTIN  "(" ")"        { $$ = new ast::Call($1, @$); }
-     | CALL_BUILTIN "(" vargs ")"   { $$ = new ast::Call($1, $3, @$); }
-     | IDENT "(" ")"                { error(@1, "Unknown function: " + $1); YYERROR;  }
-     | IDENT "(" vargs ")"          { error(@1, "Unknown function: " + $1); YYERROR;  }
-     | BUILTIN "(" ")"              { error(@1, "Unknown function: " + $1); YYERROR;  }
-     | BUILTIN "(" vargs ")"        { error(@1, "Unknown function: " + $1); YYERROR;  }
-     | STACK_MODE "(" ")"           { error(@1, "Unknown function: " + $1); YYERROR;  }
-     | STACK_MODE "(" vargs ")"     { error(@1, "Unknown function: " + $1); YYERROR;  }
-     ;
-
-map : MAP               { $$ = new ast::Map($1, @$); }
-    | MAP "[" vargs "]" { $$ = new ast::Map($1, $3, @$); }
-    ;
-
-var : VAR { $$ = new ast::Variable($1, @$); }
-    ;
-
-map_or_var : var { $$ = $1; }
-           | map { $$ = $1; }
-           ;
-
-vargs : vargs "," expr { $$ = $1; $1->push_back($3); }
-      | expr           { $$ = new ast::ExpressionList; $$->push_back($1); }
-      ;
+basic_expr:
```

Comment:
Nit: I believe it's usually called a "primary expression" -- maybe we should stick to that? No strong opinion here

Reply:
yeah forgot to clean that up, primary makes more sense :)